Lib/test/support/__init__.py

```diff
@@ -2579,6 +2578,10 @@ def __enter__(self):
                     self.old_modes[report_type] = old_mode, old_file
 
         else:
+            try:
+                import resource
+            except ImportError:
+                resource = None
```

Comment:
Hum, can you please store it as an attribute? Add `self.resource = resource` and then use `self.resource` rather than `resource`.

I prefer to avoid raising a new exception `__exit__` with `import`.

Reply:
Thanks, victor. Make sense, I updated my PR.